src/pokenav.c

```diff
@@ -0,0 +1,1198 @@
+                dataPtr[0] = 4;
+            else
+            {
+                sub_81C7710();
+                dataPtr[0] = 5;
+            }
+        }
+        else if (v1 != 0)
+        {
+            sub_81C7850(v1);
+            if (sub_81C786C())
+                dataPtr[0] = 2;
+        }
+        break;
+    case 4:
+        if (!sub_81C75D4())
+            dataPtr[0] = 3;
+        break;
+    case 5:
+        if (!sub_81C7738())
+        {
+            v2 = gUnknown_0203CF40->field8 != 0;
+            sub_81C9430();
+            sub_81C7334();
+            if (v2)
+                SetMainCallback2(CB2_ReturnToFieldContinueScriptPlayMapMusic);
+            else
+                SetMainCallback2(CB2_ReturnToFieldWithOpenMenu);
+        }
+    }
+}
+
+bool32 sub_81C756C(u32 a0)
+{
+    u32 index;
+
+    index = a0 - UNKNOWN_OFFSET;
+    sub_81C75F4();
+    if (!gUnknown_0861F3EC[index][0]())
+        return FALSE;
+    if (!gUnknown_0861F3EC[index][2]())
+        return FALSE;
+    sub_81C7834(gUnknown_0861F3EC[index][3], gUnknown_0861F3EC[index][4]);
+    gUnknown_0203CF40->field0 = gUnknown_0861F3EC[index][1];
+    gUnknown_0203CF40->field4 = index;
+    return TRUE;
+}
+
+u32 sub_81C75D4(void)
+{
+    return sub_81C786C();
+}
+
+u32 sub_81C75E0(void)
+{
+    return gUnknown_0203CF40->field0();
+}
+
+void sub_81C75F4(void)
+{
+    InitKeys();
+}
+
+void IndirectSetVBlankCallback(IntrCallback callback)
+{
+    SetVBlankCallback(callback);
+}
+
+void sub_81C760C(void)
+{
+    SetVBlankCallback(sub_81C7418);
+}
+
+void *sub_81C761C(u32 index, u32 size)
+{
+    return gUnknown_0203CF40->field10[index] = (void*)Alloc(size);
+}
+
+void *sub_81C763C(u32 index)
+{
+    return gUnknown_0203CF40->field10[index];
+}
+
+void sub_81C7650(u32 index)
+{
+    if (gUnknown_0203CF40->field10[index] != NULL)
+    {
+        Free(gUnknown_0203CF40->field10[index]);
+        gUnknown_0203CF40->field10[index] = NULL;
+    }
+}
+
+u16 sub_81C767C(void)
+{
+    return gUnknown_0203CF40->field8;
+}
+
+void sub_81C7688(u16 a0)
+{
+    gUnknown_0203CF40->field8 = a0;
+}
+
+void sub_81C7694(u32 a0)
+{
+    u32 value;
+    value = a0;
+    if (a0 > 4)
+        value = 0;
+    gUnknown_0203CF40->fieldA = value;
+}
+
+u16 sub_81C76AC(void)
+{
+    return gUnknown_0203CF40->fieldA;
+}
+
+u32 sub_81C76B8(void)
+{
+    return gUnknown_0203CF40->fieldC;
+}
+
+bool32 sub_81C76C4(void)
+{
+    struct UnknownStruct_sub_81C76C4 *v1;
+
+    v1 = (struct UnknownStruct_sub_81C76C4*)sub_81C761C(0, sizeof(struct UnknownStruct_sub_81C76C4));
+    if (v1 == NULL)
+        return FALSE;
+    else
+    {
+        ResetSpriteData();
+        FreeAllSpritePalettes();
+        v1->data[3] = sub_81C7078(sub_81C7764, 1);
+        return TRUE;
+    }
+}
+
+u32 sub_81C76FC(void)
+{
+    u32 *v1;
+    v1 = sub_81C763C(0);
+    return sub_81C70D8(v1[3]);
+}
+
+void sub_81C7710(void)
+{
+    PlaySE(SE_PN_OFF);
+    sub_81CAADC();
+    BeginNormalPaletteFade(-1, -1, 0, 16, 0);
```

Comment:
First param should be 0xFFFFFFFF (it's a bitfield) and last param should be RGB_BLACK